machine.go

```diff
@@ -588,6 +597,8 @@ func (h *Headscale) DeleteMachine(machine *Machine) error {
 		return err
 	}
 
+	h.setLastStateChangeToNow()
```

Comment:
I'm not so sure about this change, but I was surprised to see `setLastStateChangeToNow` called in all other changes to machines, but not in `DeleteMachine`

Reply:
Yes makes sense, it kind of highlight how fragile the current "remember to update the state" approach is.